app/controllers/api/functions.php

```diff
@@ -472,6 +523,18 @@
         if (!$device->upload($file['tmp_name'], $path)) { // TODO deprecate 'upload' and replace with 'move'
             throw new Exception('Failed moving file', 500);
         }
+
+        if ($automaticDeploy === 'true') {
```

Comment:
Why `'true'` not `true`? 🤔

Reply:
HTTP params all come as string. May be you should cast it `(bool) $automaticDeploy`